aiohttp/client.py

```diff
@@ -444,11 +445,12 @@ async def _request(
                     if proxy is not None:
                         proxy = URL(proxy)
                     elif self._trust_env:
-                        for scheme, proxy_info in proxies_from_env().items():
-                            if scheme == url.scheme:
-                                proxy = proxy_info.proxy
-                                proxy_auth = proxy_info.proxy_auth
-                                break
+                        if not proxy_bypass(str(url.host)):
+                            for scheme, proxy_info in proxies_from_env().items():
```

Comment:
I feel like determining the allowed proxies should be moved to a helper function or maybe inside of `proxies_from_env()`. I suggest having a new param `for_url` that, if passed, would act as a guard. It'd look like
```python
def proxies_from_env(for_url: yarl.URL = None):
    if for_url is not None and proxy_bypass(for_url.host):
        return {}

    ...
```

Reply:
Idk, that would changes what `proxies_from_env()`'s name suggests it's doing. The way it is now it reads
if url should be proxied; get the proxy from env

Maybe pull the logic out into a new helper function?
```
def get_env_proxy_for_url(url: yarl.URL = None):
    proxy = None
    proxy_auth = None

    if not proxy_bypass(url.host):
        for scheme, proxy_info in proxies_from_env().items():
            if scheme == url.scheme:
                proxy = proxy_info.proxy
                proxy_auth = proxy_info.proxy_auth
                break

    return (proxy, proxy_auth)
```

Reply:
Yeah, maybe a new helper would be good. Just don't do this much nesting please (also don't forget a `None`-guard:
```python
def get_env_proxy_for_url(url: yarl.URL = None):
    if for_url is not None and proxy_bypass(for_url.host):
        return None, None

    for scheme, proxy_info in proxies_from_env().items():
        if scheme == url.scheme:
            return proxy_info.proxy, proxy_info.proxy_auth

    return None, None
```

Reply:
> Just don't do this much nesting please

What's this in reference to?
My original PR was just adding an extra if condition. I could refactor some of the code if that's what you mean?

Reply:
I was talking about the snippet in your previous comment, it has if->for->if->break->return flow that could've been more straightforward if rewritten as shown in my last comment.

Reply:
Oh yeah, sorry first time contributing to this repo, I wasn't sure how much refactoring would be okay.

I changed the code to be more straightforward since it was pulled out into a new function. Let me know if you want me to change it back to what's in your last comment.

Reply:
No problem, that's why we have a review process in place :) I've added a code suggestion in the recent review with a bunch of comments on why I think those changes are necessary.